### ISSUE TITLE
Bloc isn't updated with new params within BlocProvider

### ISSUE THREAD

**Issue by dened** [felangel/bloc#2173]
I'm not sure it is a bug or a feature)

I know that isn't allowed to use Provider.of(context) inside BlocProvider(create) -  or using another changeable value. But I want to recreate a bloc when my model changes.
I tried to use a workaround like this:

```dart
Builder(
  builder: (context) {
    final user = Session.of(context).user;
    return BlocProvider<SimpleBloc>(
      create: (context) => SimpleBloc(user),
      child: BlocBuilder<SimpleBloc, SimpleState>(
        builder: (context, state) => state.when(initial: (info) => Text(info)),
      ),
    );
  }
),
```
But SimpleBloc is created only once and don't get a new value of user

**Comment by felangel** [comment]
Hi @dened 👋 
Thanks for opening an issue!

I believe this is a duplicate of https://github.com/felangel/bloc/issues/2127. Closing for now and feel free to follow the discussion there 👍

**Comment by orkun1675** [comment]
Hi @felangel thanks for your work on #2127, I am able to use BlocBuilder to watch for new Bloc instances thanks to it.

But I don't think this issue is a duplicate of #2127. I can't use BlocProvider to replace the Bloc instance, it will only create one Bloc and continue using that even when the create() method is updated.

Doesn't work:
```
child: BlocProvider(
  create: (context) => ExhibitBloc(
      playerRepository: context.read<PlayerRepository>(),
      artwork: museumPieces.filteredArtworks[index]),
  child: LibraryItem(),
)
```

Works:
```
child: BlocProvider.value(
  value: bloc,
  child: LibraryItem(),
)
```

I believe this is WAI, right?

Could we potentially add support for re-creating blocs if the dependencies change? Reasoning being very similar to #2127 where children Blocs are only concerned with displaying given state and don't need to know that they are part of a larger list.

**Comment by orkun1675** [comment]
Following up, after reading #2124 I was able to get this working by using the Provider package:

```
child: ProxyProvider<MuseumPiecesBloc, ExhibitBloc>(
  create: (context) => ExhibitBloc(
      playerRepository:
          context.read<PlayerRepository>(),
      artwork: museumPieces.filteredArtworks[index]),
  update: (context, MuseumPiecesBloc museumPiecesBloc,
          previous) =>
      ExhibitBloc(
          playerRepository:
              context.read<PlayerRepository>(),
          artwork: museumPiecesBloc
              .state.filteredArtworks[index]),
  child: LibraryItem(),
)
```

Obviously mixing Provider and Bloc doesn't feel too clean. Is it safe to do so? Should we state that this is okay to do in the Bloc docs if we'd rather not implement a BlocProxyProvider?
